sdk/core/azure-core-amqp/inc/azure/core/amqp/connection.hpp

```diff
@@ -114,59 +168,206 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
     virtual void OnIoError(Connection const& connection) = 0;
   };
 
+  /** @brief Options used to create a connection. */
+  struct ConnectionOptions
+  {
+    /** @brief The idle timeout for the connection.
+     *
+     * If no frames are received within the timeout, the connection will be closed.
+     */
+    std::chrono::milliseconds IdleTimeout{std::chrono::milliseconds(60000)};
+
+    /** @brief The maximum frame size for the connection.
+     *
+     * The maximum frame size is the largest frame that can be received on the connection. During
+     * the initial connection negotiation, each peer will send a max frame size. The smaller of the
+     * two values will be used as the maximum frame size for the connection.
+     *
+     * @remarks The maximum frame size must be at least 512 bytes. The default value is the maximum
+     * value for a uint32.
+     */
+    uint32_t MaxFrameSize{std::numeric_limits<uint32_t>::max()};
+
+    /** @brief The maximum number of channels supported.
+     *
+     * A single connection may have multiple independent sessions active simultaneously up to the
+     * negotiated maximum channel count.
+     */
+    uint16_t MaxChannelCount{65535};
+
+    /** @brief Properties for the connection.
+     *
+     * The properties map contains a set of fields intended to indicate information about the
+     * connection and its container.
+     *
+     */
+    Azure::Core::Amqp::Models::AmqpMap Properties;
+
+    /** @brief Port used to communicate with server.
+     *
+     * @remarks The default port is the AMQP TLS Port (5671). Ports other than the default will not
+     * use TLS to communicate with the service.
+     */
+    uint16_t Port{AmqpTlsPort};
+
+    /**
+     * Note that the AMQP specification defines the following fields in the open performative which
+     * are not supported by the underlying uAMQP stack:
+     *
+     * - outgoing-locales
+     * - incoming-locales
+     * - offered-capabilities
+     * - desired-capabilities
+     *
+     */
+
+    /** @brief Defines the ID of the container for this connection. If empty, a unique 128 bit value
+     * will be used.
+     */
+    std::string ContainerId;
+
+    /** @brief Enable tracing from the uAMQP stack.
+     */
+    bool EnableTrace{false};
+  };
+
   class Connection final {
   public:
+    /** @brief Construct a new AMQP Connection.
+     *
+     * @param hostName The name of the host to connect to.
+     * @param options The options to use when creating the connection.
+     * @param eventHandler The event handler for the connection.
+     *
+     * @remarks The requestUri must be a valid AMQP URI.
+     */
     Connection(
-        std::shared_ptr<Azure::Core::Amqp::Network::_internal::Transport> transport,
+        std::string const& hostName,
         ConnectionOptions const& options,
         ConnectionEvents* eventHandler = nullptr);
```

Comment:
It can't be because that would force the event handlers to all be const, and that is too much of a stretch for the event handlers (the whole point of the event handlers is to allow them to react to the events).

Good idea though.